### ISSUE TITLE
How are we going to make the switch from .com to .org?

### ISSUE THREAD

**Issue by jfly** [cubingusa/org-old#9]
@jbcm627 says:

> I was implicitly assuming .org content would all go live at once when ready, and .com pages would redirect. A simple solution would be to just remove those pages from .com, and redirect all 404 cubingusa traffic to .org. Then on the .org side (Jekyll), use something like this:
https://github.com/jekyll/jekyll-redirect-from

"those pages" refers to:

- https://www.cubingusa.com/index.php
- https://www.cubingusa.com/cubers.php
- https://www.cubingusa.com/ctutorial.php and sub-pages
- https://www.cubingusa.com/cguide.php and sub-pages
- https://www.cubingusa.com/about.php

We currently have a redirect from cubingusa.org to cubingusa.com. @jbcm627, in your proposal above, you don't suggest how we'd handle that. Would we remove the redirect entirely? (that seems reasonable to me, but maybe there's some subtlety I'm not seeing)

**Comment by jbmertens** [comment]
> We currently have a redirect from cubingusa.org to cubingusa.com. @jbcm627, in your proposal above, you don't suggest how we'd handle that. Would we remove the redirect entirely? (that seems reasonable to me, but maybe there's some subtlety I'm not seeing)

Agreed, the redirect should be removed. (Sorry, again I was implicitly assuming this.)

Related, the above proposal can also handle competition websites (#5). In order to keep the directory structure in /org clean, I'd like to propose:
 - Put all static-ified competition website content from #5 in a subdirectory, let's call it 'legacy'.
 - Have .com redirect all 404 traffic from .com/[URI] to .org/legacy/[URI]
 - Have Jekyll explicitly redirect the pages mentioned above from .org/legacy/[URI] to their replacements.
 - Remove content from .com once a .org replacement is in place.

**Comment by jfly** [comment]
> Have .com redirect all 404 traffic from .com/[PATH] to .org/legacy/[PATH]

I like the cleanliness of this idea, but I think it might prove problematic if you ask yourself "in the fullness of time, what do we plan to do with the cubingusa.com domain?" Currently, the dot com domain is served up by an EC2 instance, so we can definitely implement this .com/[PATH] to .org/legacy/[PATH] redirect you propose. However, in the fullness of time, I think we'd like to get rid of this EC2 instance, and then our options become a bit more limited. Two ideas:

1. We use namecheap for hosting both our domains, so we could set up a redirect on their site to go from .com/[PATH] to .org/[PATH]. It's not clear to me from reading [their redirect documentation](https://www.namecheap.com/support/knowledgebase/article.aspx/385/77/how-do-i-setup-url-forwarding-for-a-domain) if it's possible to inject a `/legacy` into that url when doing the redirect. This would require some experimentation.
2. Set up github pages for the dot com domain, and have it serve up a html file with a redirect like the [jekyll-redirect-from library does](https://github.com/jekyll/jekyll-redirect-from/blob/master/lib/jekyll-redirect-from/redirect.html). This would require us to either list all urls up front, or do something clever with javascript.
3. Something clever I'm not thinking of? Maybe using AWS gateway and lambda?

**Comment by jbmertens** [comment]
Looking at looks like namecheap page suggests to me it is possible. I see a bit about redirecting to subdirs, etc, around the text here:

> Redirect a domain/subdomain to a specific page on a website and pass values via query strings (for domains pointed to our BasicDNS)

Hopefully that will be sufficient long-term.

**Comment by jbmertens** [comment]
As an update after chatting with @jfly and @coder13, it looks like the namecheap URL redirect idea will work. As a test, I tried directing cubingusa.org/* to cubingusa.com/test/*, with the namecheap configuration seen below:

![capture](https://cloud.githubusercontent.com/assets/964048/26366971/0b7cb4ee-3fbb-11e7-9074-3ac81cf1f8e0.PNG)

This successfully redirected me from 
www.cubingusa.org/hello.foo
to
www.cubingusa.com/test/hello.foo.
(Note that the redirect is now reverted to not add in the /test.) Hopefully we can set this up in reverse when it comes time to switch permanently. We may also want to use a wildcard redirect, so the former subdomains koii.cubingusa.com/*, caltech.cubingusa.com, etc, are all correctly redirected.

**Comment by jfly** [comment]
Great news! So we can definitely proceed with putting everything old under a "legacy" directory.
